core/state_processor.go

```diff
@@ -185,7 +185,7 @@ func InsertBlockHashHistoryAtEip2935Fork(statedb *state.StateDB, prevNumber uint
 	statedb.Witness().TouchFullAccount(params.HistoryStorageAddress[:], true)
 
 	ancestor := chain.GetHeader(prevHash, prevNumber)
-	for i := prevNumber; i > 0 && i >= prevNumber-params.Eip2935BlockHashHistorySize; i-- {
+	for i := prevNumber; i > 0 && i > prevNumber-params.Eip2935BlockHashHistorySize; i-- {
```

Comment:
Can be removed when https://github.com/gballet/go-ethereum/pull/438 is merged and the base branch change bubbles up when we rebase branches. For now leaving the fix since it's needed.